core/lib/basic_types/src/lib.rs

```diff
@@ -162,6 +149,12 @@ impl TryFrom<u64> for L2ChainId {
     }
 }
 
+impl From<u32> for L2ChainId {
```

Comment:
Why only u32? I think we have to have TryFrom<u64> as well

Reply:
why?
2^32 is < then MAX

Reply:
GitHub ate the `u64`